posts/test_post_uploading.py

```diff
@@ -0,0 +1,126 @@
+"""Unit tests for posts uploading."""
+
+# Authors: mukobi
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import mongomock
+import app
+
+VALID_POST_FULL = {
+    "event_id": "abc123",
+    "author_id": "kurt_vonnegut",
+    "text": "This is a very valid post.",
+    "files": [
+        "Pretend I am a long base-64 encoded file",
+        "Another pretend encoded file"
+    ]}
+VALID_POST_TEXT_NO_FILES = {
+    "event_id": "abc123",
+    "author_id": "ray_bradbury",
+    "text": "This post has no files but is still valid.",
+    "files": []}
+VALID_POST_FILES_NO_TEXT = {
+    "event_id": "abc123",
+    "author_id": "douglas_adams",
+    "text": "",
+    "files": [
+        "This post has no text but is valid because it has at least one file."
+    ]}
+INVALID_POST_NO_TEXT_NOR_FILES = {
+    "event_id": "abc123",
+    "author_id": "jrr_tolkien",
+    "text": "",
+    "files": []}
+INVALID_POST_NOT_ENOUGH_ATTRS = {
+    "where_did_all_the_attributes_go?": "I don't know"}
+INVALD_POST_TOO_MANY_ATTRS = {
+    "event_id": "abc123",
+    "author_id": "grr_martin",
+    "text": "I have a valid description but too many fields.",
+    "files": [],
+    "woah_where_did_I_come_from?": "malicious data"}
+
+
+class TestPostUploading(unittest.TestCase):
+    """Test post uploading of users service."""
+    @staticmethod
+    def posts_are_equal(first, second):
+        """Returns whether the 2 posts have the same data.
+
+        Only looks at valid post attributes
+        (event_id, author_id, text, files).
+        """
+        return (first["event_id"] == second["event_id"] and
+                first["author_id"] == second["author_id"] and
+                first["text"] == second["text"] and
+                first["files"] == second["files"])
+
+    def setUp(self):
+        """Set up mock DB for testing"""
+        self.mock_collection = mongomock.MongoClient().db.collection
+
+    def test_full_upload(self):
+        """Can upload a full post object with both text and files."""
+        app.upload_new_post_to_db(VALID_POST_FULL, self.mock_collection)
+        post_in_db = self.mock_collection.find_one({})
+        self.assertIsNotNone(post_in_db)
+        self.assertTrue(self.posts_are_equal(VALID_POST_FULL, post_in_db))
```

Comment:
Consider making `posts_are_equal` a classmethod like `assertPostsEqual()` which reports both values when they are not equal.

Reply:
Done https://github.com/knative-portability/large-events/pull/16/commits/9020e0fd725fd4c8fe9bb1d0695f6348e2b3936f